sdk/dotnet/Pulumi/Deployment/Deployment_Run.cs

```diff
@@ -184,22 +194,18 @@ private static async Task<ImmutableArray<Resource>> TestAsync(IMocks mocks, Func
         // this method *must* remain marked async
         // in order to protect the scope of the AsyncLocal Deployment.Instance we cannot elide the task (return it early)
         // if the task is returned early and not awaited, than it is possible for any code that runs before the eventual await
-        // to be executed synchronously and thus have multiple calls to one of the Run methods affecting eachothers Deployment.Instance
+        // to be executed synchronously and thus have multiple calls to one of the Run methods affecting each others Deployment.Instance
         internal static async Task<int> CreateRunnerAndRunAsync(
             Func<Deployment> deploymentFactory,
             Func<IRunner, Task<int>> runAsync)
         {
-            var enableVerboseLogging = Environment.GetEnvironmentVariable("PULUMI_DOTNET_LOG_VERBOSE");
```

Comment:
I don't think the declaration should be pulled out of here. Having the declaration in here means that consequent runs from Automation API will get their own setting.

For instance, by doing it in the static constructor if the first run in Automation API has verbose logging enabled than all subsequent runs in the same process will have it enabled and will have no control over it since we only check the environment variable once.

Now, it looks like the way our code currently existed also had a problem - since we were just instantiating over the static serilog logger it was entirely possible that we would overwrite the logger from a currently ongoing concurrent Automation API run. So that was a problem too.

To resolve both of these issues I think we should keep the environment variable check and the instantiation in this method, and we should see if we can refactor the logger to be provided into the DeploymentInstance constructor so that it belongs only to that run.